lib/diagnostic.rb

```diff
@@ -100,20 +118,26 @@ def eat_meat(food)
 end
 
 ##
-# your response here
+class Lion < Cat
+  include Carnivorous
+  def roar
+    p 'ROAR!'
+  end
+end
 ##
 
 # #
 # What are some of the advantages of using composition (i.e. mixins)
 # over using direct inheritance?
 
 ##
-# your response as a comment here
+# Mixins allow you to draw from multiple sources/modules whereas direct inheritance
+# does not.
 ##
 
 ##
 # Fix the 'ComboAttack' class below so that calling
-# `ComboAttack.new.punch.move('left').kick.damage` doesn't cause an error.
+# `ComboAttack.new.punch.move('left').kick.damage` doesn't cause an error. <-- not sure how to do this?
 #
```

Comment:
That's okay, it's a tough question and the wording could be reworked. 

```ruby
class ComboAttack
  attr_reader :moves, :damage

  def initialize
    @moves = []
    @damage = 0
  end

  def punch
    @moves << 'punch'
    @damage += 5
    @damage *= multiplier
    self
  end

  def move(direction)
    @moves << "move #{direction}"
    self
  end

  def kick
    @moves << 'kick'
    @damage += 10
    @damage *= multiplier
    self
  end

  def self.get_possible_moves
    'kick, move, punch'
  end

  private

  def multiplier
    case moves
    when ['punch', 'move left', 'kick']
      1.5
    when ['kick', 'punch', 'up']
      2
    else
      1
    end
  end
end

```

Does this solution make sense to you?